libtransmission/peer-io.cc

```diff
@@ -677,13 +678,28 @@ void tr_peerIo::on_utp_error(int errcode)
 {
     tr_logAddTraceIo(this, fmt::format("utp_on_error -- {}", utp_error_code_names[errcode]));
 
-    if (got_error_ != nullptr)
+    if (got_error_ == nullptr)
     {
-        tr_error* error = nullptr;
+        return;
+    }
+
+    tr_error* error = nullptr;
+    switch (errcode)
+    {
+    case UTP_ECONNREFUSED:
+        tr_error_set_from_errno(&error, ECONNREFUSED);
+        break;
+    case UTP_ECONNRESET:
+        tr_error_set_from_errno(&error, ECONNRESET);
+        break;
+    case UTP_ETIMEDOUT:
+        tr_error_set_from_errno(&error, ETIMEDOUT);
+        break;
+    default:
```

Comment:
translate the utp error codes into standard errno values so that error handlers don't have to know about custom error types. This is one reason that `4.0.0` didn't retry with TCP after UTP failure